api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/ClusterEndpointProvider.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.proxy.service;
+
+import java.util.Optional;
+
+/**
+ * Provides the addresses of the network endpoints required by a virtual cluster.
+ */
+public interface ClusterEndpointProvider {
```

Comment:
Given implementations of `ClusterEndpointProvider` don't actually bind or ports or anything similar I wonder if it should actually be `ClusterEndpointMapper` or maybe `BrokerToEndpointMapper` (though given we are talking about bootstrap addresses too `Broker` seems like a poor prefix).

Reply:
I'm not happy with the name either.  I'm thinking `ClusterEndpointConfigProvider` might be better a reflection of the class's role.  It exists to tell the rest of the system how this virtual cluster needs its endpoints configured.  Do I need an exclusive port?  How should names of the broker endpoints be formed?

I'd like to defer a rename decision right now until we understand the problem space slightly better.  I'm expecting https://github.com/kroxylicious/kroxylicious/issues/265 and https://github.com/kroxylicious/kroxylicious/issues/271to shed light.

Reply:
Telling the rest of the system how endpoints should be configured vs being the thing that configures them is at the root of my difficulties with the interface.

I think calling it a config provider for now and revisiting later would be the best call.